main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2732,6 +2707,7 @@ object Defaults extends BuildCommon {
           inputs.options.sources.toVector,
           scalacOptions.value.toVector,
           javacOptions.value.toVector,
+          outputPath.value + prefix(configuration.value.name),
```

Comment:
To be even more precise, I think we should replace this with:

```suggestion
          c.toVirtualFile(inputs.options.classesDirectory)
```

If the class directory changes, we should re-compile to a new class directory.

Reply:
That's what I did in https://github.com/sbt/sbt/pull/7592/commits/9c2a36d95fc377f5366e6194fa9dfd25d7783cd7

Reply:
Addressed in b7e037adbbf3a0e33a97384849fc6803e17f0118